packages/fuels-core/src/types/tx_status.rs

```diff
@@ -177,20 +205,32 @@ impl From<ClientTransactionStatus> for TxStatus {
                 total_fee,
                 ..
             } => {
-                let revert_id = program_state
-                    .and_then(|state| match state {
-                        ProgramState::Revert(revert_id) => Some(revert_id),
-                        _ => None,
-                    })
-                    .expect("Transaction failed without a `revert_id`");
-                TxStatus::Revert(Revert {
+                let revert_id = program_state.and_then(|state| match state {
+                    ProgramState::Revert(revert_id) => Some(revert_id),
+                    _ => None,
+                });
+
+                TxStatus::Failure(Failure {
                     receipts,
                     reason,
                     revert_id,
                     total_gas,
                     total_fee,
                 })
             }
+            ClientTransactionStatus::PreconfirmationFailure {
+                reason,
+                receipts,
+                total_gas,
+                total_fee,
+                ..
+            } => TxStatus::Failure(Failure {
```

Comment:
Why not have a preconfirmation failure as well on our side?

I'm wary of doing `unwrap_or_default` instead of keeping the receipts optional if that is the right way of modeling them.

Reply:
Talked with the core guys. Currently, there is no case where receipts can be `None` in pre-confirmations so I would suggest we do not add a new struct for now. We can always add it later if need be.